components/auth/auth.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express');
+const passport = require('passport');
+const Joi = require('joi');
+
+const router = express.Router();
+
+const validateForm = (req, res, next) => {
+  const { username, password } = req.body;
+
+  const user = {
+    username,
+    password,
+  };
+
+  const schema = Joi.object().keys({
+    username: Joi.string()
+      .regex(/^[a-z0-9]{5,15}/)
+      .required(),
+    password: Joi.string()
+      .alphanum()
+      .min(6)
+      .required(),
+  });
+
+  Joi.validate(user, schema, (err, value) => {
+    if (err) {
+      return res.status(400).json(err.details[0]);
+    }
+
+    if (!value) return false;
+    return next();
+  });
+};
+
+router.post(
+  '/login',
+  validateForm,
+  (req, res) => {
+    passport.authenticate('local-login', (err, user, message, status) => {
```

Comment:
I don't understand why the custom callback has 4 parameters. Where does this come from? The documentation only mentions 3: `err`, `user` and `info`.

On a side note I am not sure why we need a custom callback at all. Can't we simply use `passport.authenticate` as middleware and add our own middleware after to return the user info (from `req.user`)?